src/transformers/image_transforms.py

```diff
@@ -0,0 +1,216 @@
+
+import numpy as np
+
+from transformers.utils.import_utils import is_tf_available, is_torch_available, is_vision_available
+
+if is_vision_available():
+    import PIL
+
+    from .image_utils import (
+        ChannelDimension,
+        get_image_size,
+        infer_channel_dimension_format,
+        is_jax_tensor,
+        is_tf_tensor,
+        is_torch_tensor,
+    )
+
+
+if TYPE_CHECKING:
+    if is_torch_available():
+        import torch
+    if is_tf_available():
+        import tensorflow as tf
+
+
+def to_channel_dimension_format(image: np.ndarray, channel_dim: Union[ChannelDimension, str]) -> np.ndarray:
+    """
+    Converts `image` to the channel dimension format specified by `channel_dim`.
+
+    Args:
+        image (`numpy.ndarray`):
+            The image to convert to the PIL Image format.
+        channel_dim (`ChannelDimension`):
+            The channel dimension format to use.
+
+    Returns:
+        image: A converted np.ndarray.
+    """
+    current_channel_dim = infer_channel_dimension_format(image)
+    target_channel_dim = ChannelDimension(channel_dim)
+    if current_channel_dim == target_channel_dim:
+        return image
+
+    if target_channel_dim == ChannelDimension.FIRST:
+        return image.transpose((2, 0, 1))
+
+    if target_channel_dim == ChannelDimension.LAST:
+        return image.transpose((1, 2, 0))
+
+    raise ValueError("Unsupported channel dimension format: {}".format(channel_dim))
+
+
+def to_pil_image(
+    image: Union[np.ndarray, PIL.Image.Image, "torch.Tensor", "tf.Tensor"], rescale=None
+) -> PIL.Image.Image:
+    """
+    Converts `image` to a PIL Image. Optionally rescales it and puts the channel dimension back as the last axis if
+    needed.
+
+    Args:
+        image (`PIL.Image.Image` or `numpy.ndarray` or `torch.Tensor`):
+            The image to convert to the PIL Image format.
+        rescale (`bool`, *optional*):
+            Whether or not to apply the scaling factor (to make pixel values integers between 0 and 255). Will default
+            to `True` if the image type is a floating type, `False` otherwise.
+    """
+    if isinstance(image, PIL.Image.Image):
+        return image
+
+    if is_torch_tensor(image) or is_tf_tensor(image):
+        image = image.numpy()
+    elif is_jax_tensor(image):
+        image = np.array(image)
+
+    if not isinstance(image, np.ndarray):
+        raise ValueError("Input image must be of type PIL.Image.Image, numpy.ndarray or torch.Tensor")
+
+    # If the channel as been moved to first dim, we put it back at the end.
+    image = to_channel_dimension_format(image, ChannelDimension.LAST)
+
+    # PIL.Image can only store uint8 values, so we rescale the image to be between 0 and 255 if needed.
+    rescale = isinstance(image.flat[0], float) if rescale is None else rescale
+    if rescale:
+        rescale = image * 255
+    image = image.astype(np.uint8)
+    return PIL.Image.fromarray(image)
+
+
+def get_resize_output_image_size(
```

Comment:
The draft PR looks good to me! I have a question though, do we need two methods for resizing (`get_resize_output_image_size` and `resize`)? 

The `default_to_square` argument is useful and I think removing it from the `resize` method might break some pipelines

Reply:
The two main reasons for having both `get_resize_output_image_size` and `resize` were:
* Separation of concerns - `resize` is only responsible for resizing an image, not shape calculations. Coming from TF I found the `resize` behaviour surprising. This forces a more explicit shape passing when processing. 
* The shape logic is quite complex, this broke it down into smaller functions. If the shape calculation were moved back inside `resize` I would probably still have put the logic outside in a private function, but this is a style preference. 

I don't feel very strongly about this and happy to move the finding shape logic to be internal to the method again. 

For `default_to_square` where do you think this might break? Now you've highlighted - I could foresee in cases when individual feature extractor methods are used, although I haven't seen this any of our pipelines or notebooks. Let me know if there's something I'm overlooking or examples.